### DISCUSSION TITLE
[npm-pulish] Does not ignore files properly

### DISCUSSION THREAD

**Discussion by taoning2014** [npm/feedback#639] [CLI] [answered]
In the npm-publish doc, there is a section talking about [Files included in package](https://docs.npmjs.com/cli/v8/commands/npm-publish#files-included-in-package). Specifically, it mentions `.gitignore` will be respected if `.npmignore` does **NOT** exist:

> If there is a .gitignore or .npmignore file, then ignored files in that and all child directories will be excluded from the package. If both files exist, then the .gitignore is ignored, and only the .npmignore is used.

However, I find this is not the case and it takes me a while to debug in my own package (thanks to @fsevenm who raised [an issue in my package](https://github.com/taoning2014/srt-validator/issues/37) to catch my attention). In fact, some of the folders that listed in `.gitignore` will **still be published** to NPM. In my Repo, I listed `/dist` in the [.gitignore](https://github.com/taoning2014/srt-validator/blob/npm-publish-issue/.gitignore), but it still get published.

**How to reproduce:**
1. `git clone -b npm-publish-issue git@github.com:taoning2014/srt-validator.git`
2. `npm install`
3. `npm run build`
4. [Testing whether your .npmignore or files config works](https://docs.npmjs.com/cli/v8/using-npm/developers#testing-whether-your-npmignore-or-files-config-works) by either run `npx npm-packlist` or `npm pack`. 

**Result:** `dist/srtValidator.js` is included in the reulst of step 4.

**System Information:** 
- Darwin Kernel Version 21.1.0
- node@16.13.1
- npm@8.1.2

**Comment by MylesBorins** [comment, marked as answer]
@taoning2014 could you please open an issue over in https://github.com/npm/cli to track this bug

**Reply by ljharb** [reply]
(also, before filing it, confirm it still happens on npm v8.5.2)
